ts/src/cex.ts

```diff
@@ -856,57 +862,59 @@ export default class cex extends Exchange {
+        let amount = this.safeString (order, 'amount');
         // sell orders can have a negative amount
         // https://github.com/ccxt/ccxt/issues/5338
         if (amount !== undefined) {
-            amount = Math.abs (amount);
+            amount = Precise.stringAbs (amount);
         }
-        const remaining = this.safeNumber2 (order, 'pending', 'remains');
-        const filled = amount - remaining;
+        const remaining = this.safeString2 (order, 'pending', 'remains');
+        const filled = Precise.stringSub (amount, remaining);
         let fee = undefined;
         let cost = undefined;
         if (market !== undefined) {
             symbol = market['symbol'];
-            const taCost = this.safeNumber (order, 'ta:' + market['quote']);
-            const ttaCost = this.safeNumber (order, 'tta:' + market['quote']);
-            cost = this.sum (taCost, ttaCost);
+            const taCost = this.safeString (order, 'ta:' + market['quote']);
+            const ttaCost = this.safeString (order, 'tta:' + market['quote']);
+            cost = Precise.stringAdd (taCost, ttaCost);
             const baseFee = 'fa:' + market['base'];
             const baseTakerFee = 'tfa:' + market['base'];
             const quoteFee = 'fa:' + market['quote'];
             const quoteTakerFee = 'tfa:' + market['quote'];
-            let feeRate = this.safeNumber (order, 'tradingFeeMaker');
+            let feeRate = this.safeString (order, 'tradingFeeMaker');
             if (!feeRate) {
-                feeRate = this.safeNumber (order, 'tradingFeeTaker', feeRate);
+                feeRate = this.safeString (order, 'tradingFeeTaker', feeRate);
             }
             if (feeRate) {
-                feeRate = feeRate / 100.0; // convert to mathematically-correct percentage coefficients: 1.0 = 100%
+                feeRate = Precise.stringDiv (feeRate, '100'); // convert to mathematically-correct percentage coefficients: 1.0 = 100%
             }
             if ((baseFee in order) || (baseTakerFee in order)) {
                 const baseFeeCost = this.safeNumber2 (order, baseFee, baseTakerFee);
                 fee = {
                     'currency': market['base'],
-                    'rate': feeRate,
+                    'rate': this.parseNumber (feeRate),
                     'cost': baseFeeCost,
                 };
             } else if ((quoteFee in order) || (quoteTakerFee in order)) {
                 const quoteFeeCost = this.safeNumber2 (order, quoteFee, quoteTakerFee);
                 fee = {
                     'currency': market['quote'],
-                    'rate': feeRate,
+                    'rate': this.parseNumber (feeRate),
                     'cost': quoteFeeCost,
                 };
             }
         }
         if (!cost) {
-            cost = price * filled;
+            cost = Precise.stringMul (price, filled);
         }
         const side = order['type'];
```

Comment:
everytime we stumble on things like this (obj['key']) we should replace them with as safeX call because it crashes in python if the key does not exist
```
  File "/Users/cjg/Git/ccxt10/ccxt/examples/py/cli.py", line 216, in main
    result = await method(*args)
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/cex.py", line 802, in cancel_order
    return self.extend(self.parse_order({}), {'info': response, 'type': None})
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/cex.py", line 865, in parse_order
    side = order['type']
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/cex.py", line 865, in parse_order
    side = order['type']
```